streams/test-utils/src/test/java/org/apache/kafka/streams/test/wordcount/WindowedWordCountProcessorTest.java

```diff
@@ -0,0 +1,144 @@
+import java.util.Properties;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class WindowedWordCountProcessorTest {
+    @Test
+    public void testWithInMemoryStore() {
+        final MockProcessorContext context = new MockProcessorContext();
+
+        // Create, initialize, and register the state store.
+        final WindowStore<String, Integer> store =
+            Stores.windowStoreBuilder(Stores.inMemoryWindowStore("WindowedCounts",
+                                                                 Duration.ofDays(24),
+                                                                 Duration.ofMillis(100),
+                                                                 false),
+                                      Serdes.String(),
+                                      Serdes.Integer())
+                  .withLoggingDisabled() // Changelog is not supported by MockProcessorContext.
+                  .build();
+        store.init(context, store);
+        context.register(store, null);
+
+        // Create and initialize the processor under test
+        final Processor<String, String> processor = new WindowedWordCountProcessorSupplier().get();
+        processor.init(context);
+
+        // send a record to the processor
+        context.setTimestamp(101);
+        processor.process("key", "alpha beta gamma alpha");
+
+        // send a record to the processor in a new window
+        context.setTimestamp(221);
+        processor.process("key", "gamma delta");
+
+        // note that the processor does not forward during process()
+        assertThat(context.forwarded().isEmpty(), is(true));
+
+        // now, we trigger the punctuator, which iterates over the state store and forwards the contents.
+        context.scheduledPunctuators().get(0).getPunctuator().punctuate(1_000L);
+
+        // finally, we can verify the output.
+        final Iterator<MockProcessorContext.CapturedForward> capturedForwards = context.forwarded().iterator();
+        assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[alpha@100/200]", "2")));
+        assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[beta@100/200]", "1")));
+        assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[gamma@100/200]", "1")));
+        assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[delta@200/300]", "1")));
+        assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[gamma@200/300]", "1")));
+        assertThat(capturedForwards.hasNext(), is(false));
+    }
+
+    @Test
+    public void testWithPersistentStore() throws IOException {
+        final Properties properties = new Properties();
+        properties.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, "");
+        properties.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "");
+
+        final File stateDir = Files.createTempDirectory("kafka-test-").toFile();
```

Comment:
prop: Here you could use `TestUtils.tempDirectory()`.

Reply:
Ah, good call. I'd previously located this test in the examples module where the function wasn't available.

Reply:
done!